polynomial_fit/PolynomialFit.cpp

```diff
@@ -0,0 +1,359 @@
+/** @file
+    @brief Produces a distortion polynomial and partial display description
+           from a table of display locations to angle inputs.
+
+    @date 2019
+
+    @author
+    Russ Taylor working through ReliaSolve.com for ValitXR.
+*/
+
+// Copyright 2019 ValityXR.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Standard includes
+#include <string>
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <cmath>
+#include <vector>
+#include <stdlib.h> // For exit()
+
+// Local includes
+#include "types.h"
+
+// Global constants and variables
+static bool g_verbose = false;
+static double MY_PI = (4.0*atan(1.0));
+
+// Forward declarations
+static int testAlgorithms();
+
+static void writePolynomial(std::ostream &s, std::vector<double> const &poly)
+{
+  s << "[";
+  for (size_t i = 0; i < poly.size(); i++) {
+    if (i == 0) { s << " "; }
+    else { s << ","; }
+    s << std::setprecision(4) << poly[i];
+  }
```

Comment:
Wouldn't it be more logical to emit the closing bracket within this function instead of requiring the caller to do it themselves?

Reply:
Good catch.  Added that in a new commit.